src/System.Linq.Parallel/src/System/Linq/Parallel/Channels/AsynchronousChannel.cs

```diff
@@ -64,24 +65,24 @@ internal sealed class AsynchronousChannel<T> : IDisposable
         //   1 consumer thread to access this thing concurrently. It's been carefully designed
         //   to avoid locking, but only because of this restriction...
 
-        private readonly T[][] _buffer;              // The buffer of chunks.
+        private readonly T[]?[] _buffer;              // The buffer of chunks.
         private readonly int _index;            // Index of this channel
         private volatile int _producerBufferIndex;   // Producer's current index, i.e. where to put the next chunk.
         private volatile int _consumerBufferIndex;   // Consumer's current index, i.e. where to get the next chunk.
 
         private volatile bool _done;        // Set to true once the producer is done.
 
-        private T[] _producerChunk;         // The temporary chunk being generated by the producer.
+        private T[]? _producerChunk;         // The temporary chunk being generated by the producer.
         private int _producerChunkIndex;    // A producer's index into its temporary chunk.
-        private T[] _consumerChunk;         // The temporary chunk being enumerated by the consumer.
+        private T[]? _consumerChunk;         // The temporary chunk being enumerated by the consumer.
         private int _consumerChunkIndex;    // A consumer's index into its temporary chunk.
 
         private readonly int _chunkSize;             // The number of elements that comprise a chunk.
 
         // These events are used to signal a waiting producer when the consumer dequeues, and to signal a
         // waiting consumer when the producer enqueues.
         private ManualResetEventSlim _producerEvent;
-        private IntValueEvent _consumerEvent;
+        private IntValueEvent? _consumerEvent;
```

Comment:
Why isn't _producerEvent nullable as well?

Reply:
_producerEvent instance was created newly in constructor, but consumerEvent was set from constructor argument without null check